noxfile.py

```diff
@@ -0,0 +1,110 @@
+"""
+nox configuration for cookiecutter project template.
+"""
+
+import contextlib
+import os
+import tempfile
+from pathlib import Path
+
+import nox
+
+CI = os.environ.get('CI') is not None
+
+ROOT = Path('.')
+PYTHON_VERSIONS = ['3.9']
+PYTHON_DEFAULT_VERSION = PYTHON_VERSIONS[-1]
+
+nox.options.default_venv_backend = 'venv'
+nox.options.reuse_existing_virtualenvs = True
+
+
+# In CI, use Python interpreter provided by GitHub Actions
+if CI:
+    nox.options.force_venv_backend = 'none'
+
+
+CRUFTED_PROJECT_TMPDIR = None
+
+MD_PATHS = ['*.md']
+
+
+def run_readable(session, mode="fmt"):
+    session.run(
+        'docker',
+        'run',
+        '--rm',
+        '-v', f'{ROOT.absolute()}:/data',
+        '-w', '/data',
+        '-u', f'{os.geteuid()}:{os.getegid()}',
+        'ghcr.io/bobheadxi/readable:v0.5.0@sha256:423c133e7e9ca0ac20b0ab298bd5dbfa3df09b515b34cbfbbe8944310cc8d9c9',
+        mode, *MD_PATHS
+    )
+
+
+@nox.session(name='format', python=PYTHON_DEFAULT_VERSION)
+def format_(session):
+    """Lint the code and apply fixes in-place whenever possible."""
+    session.run('pip', 'install', '-e', '.[format]')
+    session.run('ruff', 'check', '--fix', '.')
+    run_readable(session, mode="fmt")
+
+
+@nox.session(python=PYTHON_DEFAULT_VERSION)
+def lint(session):
+    """Run linters in readonly mode."""
+    session.run('pip', 'install', '-e', '.[lint]')
+    session.run('ruff', 'check', '--diff', '.')
+    run_readable(session, mode="check")
+
+
+@contextlib.contextmanager
+def crufted_project(session):
+    session.run("pip", "install", "-e", ".")
+    global CRUFTED_PROJECT_TMPDIR
+    if not CRUFTED_PROJECT_TMPDIR:
+        CRUFTED_PROJECT_TMPDIR = tempfile.TemporaryDirectory(prefix="rt-crufted_")
+        session.log("Creating project in %s", CRUFTED_PROJECT_TMPDIR.name)
+        session.run("cruft", "create", ".", "--output-dir", CRUFTED_PROJECT_TMPDIR.name, "--no-input")
+        session.notify('cleanup_crufted_project')
+    project_path = Path(CRUFTED_PROJECT_TMPDIR.name) / "project"
+    session.chdir(project_path)
+    yield project_path
+    session.chdir(ROOT)
+
+
+def rm_root_owned(session, dirpath):
+    assert not ROOT.is_relative_to(dirpath)  # sanity check before we nuke dirpath
+    session.run("docker", "run", "--rm", "-v", f"{dirpath}:/tmpdir/", "alpine:3.18.0", "rm", "-rf", "/tmpdir/project")
```

Comment:
Hm, exactly for this reason some people don't add current user to `docker` group, so the command won't work without `sudo`, and running the whole nox session with superuser rights (`sudo nox -s lint`) looks not good to me. I don't know a good solution to this, do you?

Reply:
The docker guide and our own Handbook encourages you to have docker accessible from your main user so I'm not too worried that somebody is going to do `sudo root` here - if someone did custom setup then I is up to them to work out new problems they created for themselves.

As for the problem that is here - i.e. filesystem ownership issue and potential solutin - well, there is podman... but I don't think we should be introducing it here as podman compose is not as advanced as docker compose. Additionally if we start spawning container with different uid - without a proper migration of pre-existing files (i.e. old redis/postgres databases) its going to break on startup with permission issues.

tl;dr Such a change needs more thought put into it, which is why I proposed a hacky solution for the sake of CI only right now. It has 0 impact on production.

Reply:
Ok, since we came nowhere on slack, let's leave it as you proposed